### PR TITLE
Add a metrics endpoint that provides service-level information

### PR DESCRIPTION
**Pull request by mnussbaum**
Metrics are reported from health checks in the [OpenMetrics (aka Prometheus metrics)](https://gianarb.it/blog/prometheus-openmetrics-expositon-format) format. This will enable us to use open-source tooling that already speaks the OpenMetrics format to collect stats exposed by litmus_paper.

### CODE REVIEW

**Comment by mnussbaum** [comment]
It looks like only the ruby 1.9.3 build is failing, and that is due to rvm no longer being able to install 1.9.3. Perhaps we should remove it from the build

File: lib/litmus_paper/metric/unix_socket_utilization.rb [hunk 1]
```diff
@@ -27,6 +27,15 @@ def current_health
         end
       end
 
+      def stats
+        current_health
+        {
+          :socket_active => @active,
+          :socket_queued => @queued,
+          :socket_utilization => ((@queued / @maxconn.to_f) * 100).round,
```
Comment (lollipopman) [inline lib/litmus_paper/metric/unix_socket_utilization.rb]: This method is returning the data from the instance variables of this class, that data is only as fresh as the last time this particular unicorn process was health checked, is that the behavior you want? Instead shouldn't the health checks persist the stats information to a file and then the stats methods return the latest data from the file?

Comment (mnussbaum) [reply lib/litmus_paper/metric/unix_socket_utilization.rb]: We actually refresh these instance variables immediately before they're used by invoking `current_health` on line 31. Given the rate at which we expect this endpoint to be hit we're also not concerned about overloading the system refreshing these metrics on each call.

I think the data-freshness levels as we have them are OK

**Comment by lollipopman** [comment]
@mnussbaum given that the metrics route is running a health check for every request you should use the same caching methodology as used in the `/:service/status` route. That way for checks which are expensive a user can enable caching, by setting a positive cache_ttl, and be assured that both the status route and the metrics route will not negatively effect performance.

File: lib/litmus_paper/metric/unix_socket_utilization.rb (hunk 1 above)
Comment (mnussbaum) [reply lib/litmus_paper/metric/unix_socket_utilization.rb]: I've removed the instance var setting to make it a bit more clear when values are being fetched

**Comment by mnussbaum** [comment]
@lollipopman I've added file-based caching to the metrics endpoint. Let me know if there are any other changes you'd like to see!

**Pull request merged**